### DISCUSSION TITLE
Would the Android Air's runtime ever be able to run AS2 code?

### DISCUSSION THREAD

**Discussion by NamesMT** [airsdk/Adobe-Runtime-Support#592] [Q&A]
As in title!, would be amazing to port old flash games/apps to android.

**Comment by ajwfrost** [comment]
Have you tried using a simple AS3 file as the initial content and then loading in the AS2-based SWF afterwards? My understanding is that the restriction is mostly around the start-up code, but the AIR runtime still includes the AS2 interpreter I think..
Likely to be restricted on iOS of course though, interpreted code isn't allowed there (unless you can get Apple's permission!).

**Comment by NamesMT** [comment]
> Have you tried using a simple AS3 file as the initial content and then loading in the AS2-based SWF afterwards? My understanding is that the restriction is mostly around the start-up code, but the AIR runtime still includes the AS2 interpreter I think..
> Likely to be restricted on iOS of course though, interpreted code isn't allowed there (unless you can get Apple's permission!).

Yes I did, did works for desktop, but unfortunately it isn't supported for mobile, as in this answer here linked:
[https://stackoverflow.com/a/6729443](https://stackoverflow.com/a/6729443)
and this capabilities profile table:
[https://help.adobe.com/en_US/air/build/WS144092a96ffef7cc16ddeea2126bb46b82f-8000.html](https://help.adobe.com/en_US/air/build/WS144092a96ffef7cc16ddeea2126bb46b82f-8000.html)
and various other google answers :'(
